api/v1alpha1/types.go

```diff
@@ -60,6 +60,33 @@ type CommonThanosFields struct {
 	LogFormat *string `json:"logFormat,omitempty"`
 }
 
+type Additional struct {
+	// Additional arguments to pass to the Thanos components.
+	// +kubebuilder:validation:Optional
+	AdditionalArgs []string `json:"additionalArgs,omitempty"`
+	// Additional containers to add to the Thanos components.
+	// +kubebuilder:validation:Optional
+	AdditionalContainers []corev1.Container `json:"additionalContainers,omitempty"`
+	// Additional volumes to add to the Thanos components.
+	// +kubebuilder:validation:Optional
+	AdditionalVolumes []corev1.Volume `json:"additionalVolumes,omitempty"`
+	// Additional volume mounts to add to the Thanos component container in a Deployment or StatefulSet
+	// controlled by the operator.
+	// +kubebuilder:validation:Optional
+	AdditionalVolumeMounts []corev1.VolumeMount `json:"additionalVolumeMounts,omitempty"`
+	// Additional ports to expose on the Thanos component container in a Deployment or StatefulSet
+	// controlled by the operator.
+	// +kubebuilder:validation:Optional
+	AdditionalPorts []corev1.ContainerPort `json:"additionalPorts,omitempty"`
+	// Additional environment variables to add to the Thanos component container in a Deployment or StatefulSet
+	// controlled by the operator.
+	// +kubebuilder:validation:Optional
+	AdditionalEnv []corev1.EnvVar `json:"additionalEnv,omitempty"`
+	// AdditionalServicePorts are additional ports to expose on the Service for the Thanos component.
+	// +kubebuilder:validation:Optional
+	AdditionalServicePorts []corev1.ServicePort `json:"additionalServicePorts,omitempty"`
+}
```

Comment:
just my opinion, but I'd almost remove the "additional" prefix in front of these fields. 

Looking at https://prometheus-operator.dev/docs/operator/api/#monitoring.coreos.com/v1.Prometheus for example, it might be good to keep consistency with the naming conventions there and we could even directly copy the descriptions in terms of benefiting from familiarity. 

Secondary to that, following the link above I would also add the ability to specify config maps and secrets in the same way.

Not a blocker though, just my personal thoughts.

Reply:
Thanks, have removed the additional prefix, will add the Secret and ConfigMap logic + test in next PR.